File: python/clarabel/tests/test_json.py
import clarabel
import os


def test_json_solve():

    datafile = "hs35.json"

    # load and solve with settings in file
    thisdir = os.path.dirname(__file__)
    datadir = os.path.join(thisdir, "../../../examples/data/")
    file = os.path.join(datadir, datafile)

    solver = clarabel.load_from_file(file)
    solution = solver.solve()
    assert solution.status == clarabel.SolverStatus.Solved

    # load and solve with custom settings
    settings = clarabel.DefaultSettings()
    settings.max_iter = 1
    solver = clarabel.load_from_file(file, settings)
    solution = solver.solve()
    assert solution.status == clarabel.SolverStatus.MaxIterations

File: src/julia/ClarabelRs/test/test_json.jl
using Clarabel, JuMP, SparseArrays
using ClarabelRs

# try loading a test problem with/without settings 
filename = "../../../../examples/data/hs35.json"

# with settings in file 
solver = ClarabelRs.load_from_file(filename)
solution = ClarabelRs.solve!(solver)
@assert solution.status == Clarabel.SOLVED

# with custom settings
settings = Clarabel.Settings() 
settings.max_iter = 1
solver = ClarabelRs.load_from_file(filename,settings)
solution = ClarabelRs.solve!(solver)
@assert solution.status == Clarabel.MAX_ITERATIONS
